### PR TITLE
[#804] Spec: Overhaul (Second Tranche)

### PR DESCRIPTION
**Pull request by mkarg**
This PR collects a tranche of the spec overhaul organized by issue #804:
- Fails build on error
- Not creating Spec-ZIP anymore
- Nightly Builds
- Removing unnecessary parent POM

**As these are no API changes, this is a fast-track review period of just one day as per our [committer rules](https://github.com/eclipse-ee4j/jaxrs-api/wiki/Committer-Conventions#minimum-length-of-review-period-before-merge--close-of-prs-and-closing-of-issues).**

### CODE REVIEW

**Comment by mkarg** [comment]
@eclipse-ee4j/ee4j-jaxrs-committers Review please. :-)

File: Jenkinsfile [hunk 1]
```diff
@@ -25,6 +25,9 @@ pipeline {
 				dir ('examples') {
 					sh "$MVN deploy"
 				}
+                dir ('jaxrs-spec') {
```
Comment (chkal) [inline Jenkinsfile]: The file is currently using tabs for indentation and you added spaces. I prefer spaces, but let's not mix them.

Comment (mkarg) [reply Jenkinsfile]: Absolutely +1, but can somebody tell me how I can tell [Gitpod](https://gitpod.io/#https://github.com/eclipse-ee4j/jaxrs-api/pull/808) to keep the tabs...? I type tabs, the editor shows tabs, but in the file are blanks. ;-(

Comment (mkarg) [reply Jenkinsfile]: Solved. Used `vi`. 40 years old technoligy works better than 1 year old. ;-)

**Comment by mkarg** [comment]
@eclipse-ee4j/ee4j-jaxrs-committers Please review this PR. :-)

**Comment by mkarg** [comment]
@andymc12 @asoldano Please review this PR ASAP. Thanks. :-)

**Comment by mkarg** [comment]
@eclipse-ee4j/ee4j-jaxrs-committers Please review this PR. Please, please, please! :-)

**Pull request merged**